*Replace this text with your own name and a brief description of what this notebook does.*

## Assignment 10-1

We will use the 311 service request dataset from Lab 8 and 10, which contains information about all complaints and service request calls made to 311 in New York city.

Write a piece of code that computes the probability that a call is from the Bronx or Brooklyn. 

In [1]:
import matplotlib.pyplot as plt
import pandas as pd

calls = pd.read_csv("../../data/Sept15_2023_311_Service_Requests.csv")
pd.set_option('display.max_columns', None)
calls.head(30)

Unique Key            Created Date             Closed Date Agency  \
0     58823685  09/15/2023 11:59:57 PM  09/18/2023 11:39:38 AM  DOHMH   
1     58821637  09/15/2023 11:59:44 PM  09/16/2023 12:25:10 AM   NYPD   
2     58822468  09/15/2023 11:59:34 PM  09/16/2023 12:28:09 AM   NYPD   
3     58822498  09/15/2023 11:59:32 PM  09/16/2023 12:12:27 AM   NYPD   
4     58823234  09/15/2023 11:59:29 PM  09/16/2023 12:13:37 AM   NYPD   
5     58825105  09/15/2023 11:59:25 PM  09/16/2023 12:55:33 AM   NYPD   
6     58821688  09/15/2023 11:59:22 PM  09/16/2023 01:33:56 AM   NYPD   
7     58820869  09/15/2023 11:59:19 PM  09/16/2023 12:32:55 AM   NYPD   
8     58822319  09/15/2023 11:59:14 PM  09/16/2023 01:41:05 AM   NYPD   
9     58825057  09/15/2023 11:59:11 PM  09/16/2023 12:28:40 AM   NYPD   
10    58824143  09/15/2023 11:59:05 PM  09/16/2023 12:19:49 AM   NYPD   
11    58823501  09/15/2023 11:59:02 PM                     NaN    HPD   
12    58825284  09/15/2023 11:59:01 PM  09/23/2023 10:38:38 AM    HPD   
13    58824254  09/15/2023 11:58:53 PM  09/16/2023 12:23:38 AM   NYPD   
14    58819840  09/15/2023 11:58:52 PM  09/16/2023 12:14:06 AM   NYPD   
15    58819770  09/15/2023 11:58:51 PM  09/16/2023 01:55:29 AM   NYPD   
16    58825898  09/15/2023 11:58:37 PM  09/16/2023 12:14:20 AM   NYPD   
17    58823430  09/15/2023 11:58:32 PM  09/16/2023 12:16:33 AM   NYPD   
18    58825021  09/15/2023 11:58:20 PM  09/16/2023 12:00:29 AM   NYPD   
19    58823434  09/15/2023 11:58:15 PM  09/16/2023 12:14:58 AM   NYPD   
20    58820803  09/15/2023 11:58:13 PM  09/16/2023 12:14:47 AM   NYPD   
21    58824611  09/15/2023 11:58:01 PM                     NaN   DSNY   
22    58821665  09/15/2023 11:58:01 PM  09/16/2023 12:22:35 AM   NYPD   
23    58820743  09/15/2023 11:57:55 PM  09/16/2023 12:25:18 AM   NYPD   
24    58823329  09/15/2023 11:57:51 PM  09/16/2023 12:56:32 AM   NYPD   
25    58821716  09/15/2023 11:57:40 PM  09/16/2023 12:13:42 AM   NYPD   
26    58824938  09/15/2023 11:57:33 PM  09/16/2023 12:41:21 AM   NYPD   
27    58825019  09/15/2023 11:57:31 PM  09/16/2023 12:50:34 AM   NYPD   
28    58820764  09/15/2023 11:57:22 PM  09/16/2023 12:31:52 AM   NYPD   
29    58825463  09/15/2023 11:57:19 PM  09/18/2023 12:20:35 PM   DCWP   

                                          Agency Name  \
0             Department of Health and Mental Hygiene   
1                     New York City Police Department   
2                     New York City Police Department   
3                     New York City Police Department   
4                     New York City Police Department   
5                     New York City Police Department   
6                     New York City Police Department   
7                     New York City Police Department   
8                     New York City Police Department   
9                     New York City Police Department   
10                    New York City Police Department   
11  Department of Housing Preservation and Develop...   
12  Department of Housing Preservation and Develop...   
13                    New York City Police Department   
14                    New York City Police Department   
15                    New York City Police Department   
16                    New York City Police Department   
17                    New York City Police Department   
18                    New York City Police Department   
19                    New York City Police Department   
20                    New York City Police Department   
21                           Department of Sanitation   
22                    New York City Police Department   
23                    New York City Police Department   
24                    New York City Police Department   
25                    New York City Police Department   
26                    New York City Police Department   
27                    New York City Police Department   
28                    New York City Police Department   
29       Department of Consume

In [2]:
# Compute the number of calls from Manhattan or Staten Island.
bronx_filter = calls["Borough"] == "BRONX"
brooklyn_filter = calls["Borough"] == "BROOKLYN"

# Create a dataset of only calls from Manhattan or Staten Island
bronx_or_brooklyn_calls = calls[bronx_filter | brooklyn_filter]
bronx_or_brooklyn_calls.head()

Unique Key            Created Date             Closed Date Agency  \
0     58823685  09/15/2023 11:59:57 PM  09/18/2023 11:39:38 AM  DOHMH   
3     58822498  09/15/2023 11:59:32 PM  09/16/2023 12:12:27 AM   NYPD   
4     58823234  09/15/2023 11:59:29 PM  09/16/2023 12:13:37 AM   NYPD   
6     58821688  09/15/2023 11:59:22 PM  09/16/2023 01:33:56 AM   NYPD   
10    58824143  09/15/2023 11:59:05 PM  09/16/2023 12:19:49 AM   NYPD   

                                Agency Name           Complaint Type  \
0   Department of Health and Mental Hygiene                   Rodent   
3           New York City Police Department      Noise - Residential   
4           New York City Police Department       Noise - Commercial   
6           New York City Police Department  Noise - Street/Sidewalk   
10          New York City Police Department       Noise - Commercial   

          Descriptor               Location Type  Incident Zip  \
0       Rat Sighting     3+ Family Apt. Building       11213.0   
3   Loud Music/Party  Residential Building/House       11236.0   
4   Loud Music/Party         Club/Bar/Restaurant       11226.0   
6   Loud Music/Party             Street/Sidewalk       10459.0   
10  Loud Music/Party            Store/Commercial       11222.0   

       Incident Address     Street Name      Cross Street 1    Cross Street 2  \
0     1702 UNION STREET    UNION STREET  SCHENECTADY AVENUE      UTICA AVENUE   
3    9602 FARRAGUT ROAD   FARRAGUT ROAD    EAST   96 STREET  ROCKAWAY PARKWAY   
4   1602 NEWKIRK AVENUE  NEWKIRK AVENUE    EAST   16 STREET  EAST   17 STREET   
6     1278 UNION AVENUE    UNION AVENUE    EAST  169 STREET      RITTER PLACE   
10     1 BEDFORD AVENUE  BEDFORD AVENUE    MANHATTAN AVENUE    LORIMER STREET   

   Intersection Street 1 Intersection Street 2 Address Type      City  \
0     SCHENECTADY AVENUE          UTICA AVENUE      ADDRESS  BROOKLYN   
3       EAST   96 STREET      ROCKAWAY PARKWAY      ADDRESS  BROOKLYN   
4       EAST   16 STREET      EAST   17 STREET      ADDRESS  BROOKLYN   
6       EAST  169 STREET          RITTER PLACE      ADDRESS     BRONX   
10      MANHATTAN AVENUE        LORIMER STREET      ADDRESS  BROOKLYN   

          Landmark Facility Type  Status Due Date  \
0     UNION STREET           NaN  Closed      NaN   
3    FARRAGUT ROAD           NaN  Closed      NaN   
4   NEWKIRK AVENUE           NaN  Closed      NaN   
6     UNION AVENUE           NaN  Closed      NaN   
10  BEDFORD AVENUE           NaN  Closed      NaN   

                               Resolution Description  \
0   This service request was closed because the De...   
3   The Police Department responded to the complai...   
4   This complaint does not fall under the Police ...   
6   The Police Department responded to the complai...   
10  The Police Department responded to the complai...   

   Resolution Action Updated Date Community Board           BBL   Borough  \
0          09/15/2023 11:59:57 PM     09 BROOKLYN  3.014020e+09  BROOKLYN   
3          09/16/2023 12:12:29 AM     18 BROOKLYN  3.081650e+09  BROOKLYN   
4          09/16/2023 12:13:43 AM     14 BROOKLYN  3.052370e+09  BROOKLYN   
6          09/16/2023 01:34:00 AM        03 BRONX  2.029680e+09     BRONX   
10         09/16/2023 12:19:53 AM     01 BROOKLYN  3.026460e+09  BROOKLYN   

    X Coordinate (State Plane)  Y Coordinate (State Plane)  \
0                    1003082.0                    182605.0   
3                    1010713.0                    174716.0   
4                     994706.0                    170989.0   
6                    1012302.0                    241904.0   
10                    997813.0                    203054.0   

   Open Data Channel Type Park Facility Name Park Borough Vehicle Type  \
0                  MOBILE        Unspecified     BROOKLYN          NaN   
3                  ONLINE        Unspecified     BROOKLYN          NaN   
4                   PHONE        Unspecified     BROOKLYN          NaN   
6              

In [3]:
num_bronx_or_brooklyn_calls = bronx_or_brooklyn_calls.shape[0]

In [4]:
# Compute the total number of calls
num_calls = calls.shape[0]
num_calls

9530

In [5]:
# Divide the number of calls from the Manhattan or Staten Island by the total number of calls
# to get the probability that a call is from Manhattan or Staten Island.
num_bronx_or_brooklyn_calls/num_calls

0.34900314795383003

1. What is the probability that call is from the Bronx or Brooklyn?  (Round you answer to 3 decimal places.  That is, your answer should only have three digits after the decimal point.)
2. What percentage of calls are from the Bronx or Brooklyn?

Write your answers below.

1. 0.349
2. 34.9%